aiida/cmdline/commands/profile.py

```diff
@@ -135,3 +136,71 @@ def profile_list(self, *args):
                 bold_sequence, profile, default_str, nobold_sequence, end_color)
 
+    def profile_delete(self, *args):
+        """ Deletes profile
+
+        Asks whether to delete associated database and associated database
+        user.
+        """
+        from aiida.cmdline.verdilib import Quicksetup
+        from aiida.common.setup import get_or_create_config, update_config, AIIDA_CONFIG_FOLDER
+        import os.path
+        from urlparse import urlparse
+
+        #TODO: the functions used below should be moved outside Quicksetup
+        q = Quicksetup()
+        pg_info = q._get_pg_access()
+        pg_execute = pg_info['method']
+        dbinfo = pg_info['dbinfo']
+
+        confs = get_or_create_config()
+        profiles = confs.get('profiles',{})
+        users = [ profiles[name].get('AIIDADB_USER', '') for name in profiles.keys()]
+
+        profiles_to_delete = args
+        for profile_to_delete in profiles_to_delete:
+            try:
+                profile = profiles[profile_to_delete]
+            except KeyError:
+                raise ValueError('Profile "{}" does not exist'.format(profile_to_delete))
+
+            db = profile.get('AIIDADB_NAME', '')
+            if not q._db_exists(db, pg_execute, **dbinfo):
+                print("Associated database '{}' does not exist.".format(db))
+            elif click.confirm("Delete associated database '{}'?\n" \
+                             "WARNING: All data will be lost.".format(db)):
+                print("Deleting database '{}'.".format(db))
+                q._drop_db(db, pg_execute, **dbinfo)
+
+            user = profile.get('AIIDADB_USER', '')
+            if not q._dbuser_exists(user, pg_execute, **dbinfo):
+                print("Associated database user '{}' does not exist.".format(user))
+            elif users.count(user) > 1:
+                print("Associated database user '{}' is used by other profiles "\
+                      "and will not be deleted.".format(user))
+            elif click.confirm("Delete database user '{}'?".format(user)):
+                print("Deleting user '{}'.".format(user))
+                q._drop_dbuser(user, pg_execute, **dbinfo)
+
+            repo_uri = profile.get('AIIDADB_REPOSITORY_URI','')
+            repo_path = urlparse(repo_uri).path
+            repo_path = os.path.expanduser(repo_path)
+            if not os.path.isabs(repo_path):
+                print("Associated file repository '{}' does not exist."\
+                      .format(repo_path))
+            elif not os.path.isdir(repo_path):
+                print("Associated file repository '{}' is not a directory."\
+                       .format(repo_path))
+            elif click.confirm("Delete associated file repository '{}'?\n" \
+                               "WARNING: All data will be lost.".format(repo_path)):
+                print("Deleting directory '{}'.".format(repo_path))
+                import shutil
+                shutil.rmtree(repo_path)
```

Comment:
Maybe it would make sense to already delegate the task of deleting to another function in a more obvious place which will do case handling for file system vs. object store / other future options - repositories.

Reply:
For the moment, I've created an issue. 
https://github.com/aiidateam/aiida_core/issues/695

Will use more general functionality once it becomes available.